src/main/java/com/iabtcf/v2/BitVectorTCModelV2.java

```diff
@@ -19,296 +19,352 @@
+    private Set<Integer> publisherPurposesLITransparency;
+    private Set<Integer> customPurposesConsent;
+    private Set<Integer> customPurposesLITransparency;
+
+    private final EnumSet<FieldDefs> e = EnumSet.noneOf(FieldDefs.class);
+    private final ByteBitVector bbv;
+    private Collection<ByteBitVector> remainingVectors;
+
+    private BitVectorTCModelV2(ByteBitVector bbv) {
+        this(bbv, new ByteBitVector[] {});
+    }
+
+    private BitVectorTCModelV2(ByteBitVector bbv, ByteBitVector... theRest) {
+        this.bbv = bbv;
+        this.remainingVectors = Arrays.asList(theRest);
+    }
+
+    public static BitVectorTCModelV2 fromBitVector(ByteBitVector coreBitVector, ByteBitVector... remainingVectors) {
         return new BitVectorTCModelV2(coreBitVector, remainingVectors);
     }
 
-
-
-    private int fillRemainingVector(ByteBitVector bitVector) {
-        int currentPointer = 0;
-        int segmentType = bitVector.readBits3(SEGMENT_TYPE_OFFSET);
-        currentPointer += SEGMENT_TYPE_OFFSET + SEGMENT_TYPE_LENGTH;
-        switch (segmentType) {
-            case SEGMENT_TYPE_DEFAULT:
-                break;
-            case SEGMENT_TYPE_DISCLOSED_VENDOR:
-                currentPointer = fetchSet(this.disclosedVendors, currentPointer, bitVector);
-                break;
-            case SEGMENT_TYPE_ALLOWED_VENDOR:
-                currentPointer = fetchSet(this.allowedVendors, currentPointer, bitVector);
-                break;
-            case SEGMENT_TYPE_PUBLISHER_TC:
-                currentPointer = fillPublisherPurposesTC(currentPointer, bitVector);
-                break;
+    private ByteBitVector getSegment(SegmentType segmentType) {
+        if (segmentType == SegmentType.DEFAULT) {
+            return bbv;
         }
-        return currentPointer;
-    }
-
-    private int fillPublisherPurposesTC(int currentPointer, ByteBitVector bitVector) {
-        this.publisherPurposesConsent.addAll(
-                fillSet(currentPointer, PURPOSES_CONSENT_LENGTH, bitVector));
-        currentPointer += PURPOSES_CONSENT_LENGTH;
-
-        this.publisherPurposesLITransparency.addAll(
-                fillSet(currentPointer, PURPOSE_LI_TRANSPARENCY_LENGTH, bitVector));
-        currentPointer += PURPOSE_LI_TRANSPARENCY_LENGTH;
-
-        int numberOfCustomPurposes = bitVector.readBits6(currentPointer);
-        currentPointer += TINY_INT.length();
-
-        this.customPurposesConsent.addAll(fillSet(currentPointer, numberOfCustomPurposes, bitVector));
-        currentPointer += numberOfCustomPurposes;
-
-        this.customPurposesLITransparency.addAll(
-                fillSet(currentPointer, numberOfCustomPurposes, bitVector));
-        currentPointer += numberOfCustomPurposes;
 
-        return currentPointer;
+        for (ByteBitVector rbbv : remainingVectors) {
+            int rSegmentType = rbbv.readBits3(FieldDefs.OOB_SEGMENT_TYPE.getOffset(rbbv));
+            if (rSegmentType == segmentType.ordinal()) {
```

Comment:
Relying on ordinal is generally considered bad practice. I would strongly suggest to define the values for segment types in the enum instead. The java doc for the ordinal method warns against this 

> Returns the ordinal of this enumeration constant (its position in its enum declaration, where the initial constant is assigned an ordinal of zero). Most programmers will have no use for this method. It is designed for use by sophisticated enum-based data structures, such as EnumSet and EnumMap.

Reply:
sure, i can remove the call